File: ToDoAppFrontend/to-do-app/src/Interfaces/message.ts
export default interface Message {
  id: number;
  senderId: string;
  receiverId: string;
  content: string;
  sendAt: string; // string for DateTime serialization
  isRead: boolean;
  imageUrls?: string[] | string;
  sender: {
    id: string;
    email: string;
    name?: string;
  };
  receiver: {
    id: string;
    email: string;
    name?: string;
  };
}

File: ToDoApp/EL/Models/Messages/Message.cs
using EL.Models.User;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace EL.Models.Messages
{
    public class Message
    {
        public int Id { get; set; }

        public string SenderId { get; set; } = string.Empty;
        public ApplicationUser Sender { get; set; }

        public string ReceiverId { get; set; } = string.Empty;
        public ApplicationUser Receiver { get; set; }

        public string? Content { get; set; }

        public DateTime SendAt { get; set; } = DateTime.UtcNow;

        public bool IsRead { get; set; } = false;
        public string? ImageUrls { get; set; }
    }
}
